libsolidity/experimental/codegen/IRGeneratorForStatements.cpp

```diff
@@ -322,7 +323,9 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	functionType = m_context.env->resolveRecursive(functionType);
 	m_context.enqueueFunctionDefinition(functionDefinition, functionType);
 	// TODO: account for return stack size
-	m_code << "let " << IRNames::localVariable(_functionCall) << " := " << IRNames::function(*m_context.env, *functionDefinition, functionType) << "(";
+	if (functionDefinition->returnParameterList() && !functionDefinition->returnParameterList()->parameters().empty())
+		m_code << "let " << IRNames::localVariable(_functionCall) << " := ";
+	m_code << IRNames::function(*m_context.env, *functionDefinition, functionType) << "(";
```

Comment:
I was about to say, this will just conflict with https://github.com/ethereum/solidity/pull/14620, but it looks like https://github.com/ethereum/solidity/pull/14620 doesn't account for this yet, so that's actually a good thing :-).

Reply:
Well, that PR conflicts with barely anything. I had literally zero conflicts today while rebasing it off the older version of `newAnalysis` with my fixups.

Reply:
That's not entirely true - of course it doesn't conflict unless anything significantly changes - it would very much conflict with https://github.com/ethereum/solidity/pull/14655 for example

Reply:
I meant "that PR", i.e. not this one, but the one you mentioned - #14620.

Reply:
I meant https://github.com/ethereum/solidity/pull/14620 very much will cause conflicts with https://github.com/ethereum/solidity/pull/14655